doc/user-guide/MongooseIM-High-level-Architecture.md

```diff
@@ -4,7 +4,22 @@
 
 ### Modules
 
-At its core MongooseIM is a huge message router you can customise to fit your system's needs. You can choose and enable behaviours and functionalities by configuring any of the available modules. A wide range of options includes authentication, privacy, storage, backend integration and mobile optimisations. See '[Extension Modules](../advanced-configuration/Modules.md)' for a full list.
+At its core MongooseIM is a huge message router you can customise to fit your system's needs.
+You can choose and enable behaviours and functionalities by configuring any of the available modules.
+A wide range of options includes authentication, privacy, storage, backend integration and mobile optimisations.
+See '[Extension Modules](../advanced-configuration/Modules.md)' for a full list.
+
+Modules can be configured and started either globally (for all virutal hosts served by the instance) or individually for each host.
+Modules may include dependencies on services and on other modules.
+If a module depends on other modules, required modules are started automatically.
+If a module requires certain services which are not started, the module refuses to start.
+
+### Services
+
+Some functionalities which are not related to virtual hosts but rather are provided for the whole instance or for modules started for various hosts are provided by "services".
+Services are configured globally launched on startup, before modules, so that dependencies are satisfied.
+A service can require other services to be operational; required services are started automatically.
```

Comment:
Wat? TBH I haven't noticed it in the code. Why should service be started as a dependency (with empty config!) by another service but we deny the same for modules?

Reply:
The service will not be started with empty config, it will be started with the config from the config file. Since services are singletond you cant give your own config at runtime. We do this for convenience, so that you can start a service without complete knowledge about the whole dependency tree.why we do not allow modules start required services lm not sure, it was decided on tech meeting.

Reply:
Actually the main advantage of it is that you can define services in the config file in any order and they will resolve their dependencies themselves.